### DISCUSSION TITLE
How to generate all types in client Preset

### DISCUSSION THREAD

**Discussion by liknenS** [dotansimha/graphql-code-generator#10166] [Q&A]
In client preset typescript plugin defined as:
```
{
  [`typescript`]: {
    onlyEnumTypes: true,
    onlyOperationTypes: true,
  },
}
```

But I want to generate All typescript types.

Adding typescript plugin generate duplicated types.

Any option to override typescript plugin configuration?

**Comment by Dan6erbond** [comment]
I have the same question. The old client preset generated types for everything the schema had to offer and we used these a lot for our interfaces.

**Reply by tobiasschweizer** [reply]
I have encountered the same issue when updating `@graphql-codegen/cli` to major version 5. 
I figured that one can activate a plugin `typescript` that generates all the types as before, see https://the-guild.dev/graphql/codegen/docs/advanced/how-does-it-work#example-with-graphql-codegentypescript

However, I ended up with duplicate definitions in TS (some parts are generated twice) and my conclusion is that these are two different use cases that maybe are not working well together.

**Comment by tobiasschweizer** [comment]
@liknenS Why do you want to generate all types? I actually did the same when using major version 4 but then I figured that the types I use with my React components should work with the types generated for the queries. Maybe you could write more about your use case.
